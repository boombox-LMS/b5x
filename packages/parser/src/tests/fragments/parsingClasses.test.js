// Get all the filenames in the parsingClasses directory
const fs = require("fs");
const path = require("path");
const parsingClassFolderFilenames = fs.readdirSync(path.resolve(__dirname, "../../../dist/componentBuilders/fragments/parsingClasses"));
const parsingClassFiles = parsingClassFolderFilenames.filter(filename => new RegExp(/.*.js/).test(filename));

describe("placeholder", () => { 
  parsingClassFiles.forEach(filename => {
    const { manifest } = require(`../../../dist/componentBuilders/fragments/parsingClasses/${filename}`);
    if (!manifest) {
      console.warn(`No manifest found for ${filename}, skipping parsing test for this fragment.`)
    } else {
      // parse each example markup string and test it against its api data schema
      manifest.exampleMarkupStrings.forEach((exampleMarkupString, index) => {
        test(`The example ${manifest.tagName} markup string at index ${index} matches the corresponding api data schema`, () => {
          const parsedMarkup = global.buildFragmentFromMarkupString(exampleMarkupString).packageForApi();
          const validator = () => { manifest.apiDataSchema.parse(parsedMarkup) };
          expect(validator).not.toThrowError();
          expect(parsedMarkup).toMatchSnapshot();
        });
      })
    }
  })
});