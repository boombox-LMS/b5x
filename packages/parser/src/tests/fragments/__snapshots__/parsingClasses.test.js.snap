// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder The example accordion markup string at index 0 matches the corresponding api data schema 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
    "test-fragment-uri",
    "test-fragment-uri",
  ],
  "contentType": "Accordion",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "4d8ba2b09a0224d8f7c939b743d18c47",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;

exports[`placeholder The example badge markup string at index 0 matches the corresponding api data schema 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Badge",
  "contents": "",
  "data": Object {
    "description": "Viewed the example badge in the component catalog.",
    "icon": "https://api.iconify.design/mdi/badge-outline.svg",
    "title": "Example badge",
  },
  "dependencyUris": Array [],
  "digest": "0e05c2757467bbf0b18a620da479c35a",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
